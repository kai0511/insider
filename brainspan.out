Fatal error: cannot open file 'brainspan.R': No such file or directory
