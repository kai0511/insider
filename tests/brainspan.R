##############################################
# This file models 3-D matrix factorization with additive matrix factorization using L1 penalty.
# Alternating Minimization and Coordinate Descent are employed to optimize this problem.
# For details see my note, derivartion for matrix factorization.
# This version is a mature alpha version, and some improvement is in progess.
##############################################
require(insider)

# options(error = dump_and_quit)

num_factors <- 23 

# regularization for iMF
lambda <- 6  # original value 33.7777777777778
alpha <- 0.4 # original 0.35

setwd('../results/brainspan/')
load('~/data/multi_dimensional_datasets/brainspan_dataset_annotated_fitered.RData')
dataset[is.na(dataset)] <- 0

end_idx <- 2
confounders <- dataset[ ,1:end_idx]
confounders[, 1] <- confounders[, 1] - 1
data <- log2(as.matrix(dataset[ ,-c(1:end_idx)]) + 1)

object <- insider(data, as.matrix(confounders), global_tol = 1e-10)
# object <- tune(object, latent_dimension = as.integer(num_factors), lambda = seq(1, 50, by = 5), alpha = seq(0.1, 0.6, by = 0.1))
object <- fit(object, latent_dimension = as.integer(num_factors), lambda = lambda, alpha = alpha, partition = 0)
# save(object, file = paste0("insider_brainspan_R", num_factors, "_fitted_object.RData"))
 
