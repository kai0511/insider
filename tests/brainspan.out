Loading required package: insider
number of all zero columns removed: 0 
Latent rank:  17 ---------------------------------
insider iter 0: train rmse = 2.07709300938
insider iter 0: test rmse = 2.07504227879
total_residual	44162686.8186;
row_reg_loss:	0.00030965699176;
col_reg_loss:	0.331090721085;
l1_reg_loss:	58.7830158521.
Iteration 0 ---------------------------------
insider iter 0: train rmse = 0.388084072811
insider iter 0: test rmse = 0.402085138
total_residual	1541683.54943;
row_reg_loss:	3553.03723732;
col_reg_loss:	5312.78273827;
l1_reg_loss:	5017.51132453.
Delta loss for iter 0:42607179.0523
Iteration 10 ---------------------------------
insider iter 10: train rmse = 0.340068338991
insider iter 10: test rmse = 0.352939785402
total_residual	1183793.63297;
row_reg_loss:	3732.65658911;
col_reg_loss:	4174.80862304;
l1_reg_loss:	4522.3369035.
Delta loss for iter 10:359343.445636
Iteration 20 ---------------------------------
insider iter 20: train rmse = 0.339981840199
insider iter 20: test rmse = 0.352858469394
total_residual	1183191.49697;
row_reg_loss:	3595.28207391;
col_reg_loss:	3768.82548929;
l1_reg_loss:	4298.16085031.
Delta loss for iter 20:1369.66970887
Iteration 30 ---------------------------------
insider iter 30: train rmse = 0.339958938182
insider iter 30: test rmse = 0.352822177423
total_residual	1183032.09693;
row_reg_loss:	3456.10454862;
col_reg_loss:	3505.01030618;
l1_reg_loss:	4145.40398711.
Delta loss for iter 30:715.149609036
Iteration 40 ---------------------------------
insider iter 40: train rmse = 0.339949950957
insider iter 40: test rmse = 0.352804634816
total_residual	1182969.54799;
row_reg_loss:	3316.5153607;
col_reg_loss:	3311.48834947;
l1_reg_loss:	4029.32430069.
Delta loss for iter 40:511.739770406
Iteration 50 ---------------------------------
insider iter 50: train rmse = 0.339945242337
insider iter 50: test rmse = 0.352794209658
total_residual	1182936.77778;
row_reg_loss:	3178.5636366;
col_reg_loss:	3156.81758965;
l1_reg_loss:	3933.65688003.
Delta loss for iter 50:421.06011408
Iteration 60 ---------------------------------
insider iter 60: train rmse = 0.339942308024
insider iter 60: test rmse = 0.352787046032
total_residual	1182916.3563;
row_reg_loss:	3043.66248355;
col_reg_loss:	3026.7805456;
l1_reg_loss:	3850.9754456.
Delta loss for iter 60:368.041109298
